testing/scenario_app/ios/Scenarios/ScenariosUITests/PlatformViewUITests.m

```diff
@@ -257,3 +257,40 @@ - (void)testPlatformViewWithContinuousTexture {
 }
 
 @end
+
+@interface PlatformViewScrollingUnderWidget : XCTestCase
+
+@end
+
+@implementation PlatformViewScrollingUnderWidget
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+}
+
+- (void)testPlatformViewScrollingUnderWidget {
+  XCUIApplication* app = [[XCUIApplication alloc] init];
+  app.launchArguments =
+      @[ @"--platform-view-scrolling-under-widget", @"--with-continuous-texture" ];
+  [app launch];
+
+  XCUIElement* platformView = app.textViews.firstMatch;
+  BOOL exists = [platformView waitForExistenceWithTimeout:kSecondsToWaitForPlatformView];
+  if (!exists) {
```

Comment:
can it also check for the zPosition since that seems to be the bug?

Reply:
AFAIK, XCUITest doesn't have access to the actual layers. I also did a light research and couldn't find anything online. :(

Reply:
Right, these integration tests are black box, you don't have access to the app layers, just the accessibility hooks.  A verification like that would need to happen in a XCTest unit test.

Reply:
right. If the bug is that the `zPosition` was miscalculated, then the test would need to ensure the correct values. 

I did something similar with the UIView in https://github.com/flutter/engine/blob/104e21d941fc8aed1073c6caaae71cf944837256/testing/scenario_app/ios/Scenarios/ScenariosUITests/UnobstructedPlatformViewTests.m

Reply:
I thought an incorrect z index will make the tests fail in https://github.com/flutter/engine/blob/104e21d941fc8aed1073c6caaae71cf944837256/testing/scenario_app/ios/Scenarios/ScenariosUITests/UnobstructedPlatformViewTests.m

If you don't set zPosition, are there any test failures?

Reply:
Which test would fail? I might be missing something but the tests in https://github.com/flutter/engine/blob/104e21d941fc8aed1073c6caaae71cf944837256/testing/scenario_app/ios/Scenarios/ScenariosUITests/UnobstructedPlatformViewTests.m are not checking zPosition or anything related to it right?

Reply:
There’s a comment about z index in that file. I remember the index being determined from the position of the layer in the tree I believe